src/main/java/com/fasterxml/jackson/databind/node/DecimalNode.java

```diff
@@ -57,6 +57,12 @@
     @Override
     public boolean isBigDecimal() { return true; }
 
+    @Override
+    public boolean hasFractionalPart() {
+        BigDecimal fractionalPart = _value.remainder( BigDecimal.ONE );
+        return !fractionalPart.equals(new BigDecimal("0").setScale(fractionalPart.scale()));
```

Comment:
Use `BigDecimal.ZERO` to avoid constructing a new instance?

Reply:
Actually, more specifically, create private `BigDecimal` instance to be used for comparison.

Reply:
Totally agree with you! as I know BigDecimal caches values from zero to ten. and I think the below code does the job and its not neccessary to create private BigDecimal instance:
`return !fractionalPart.equals(BigDecimal.ZERO.setScale(fractionalPart.scale()));`
am I correct?

Reply:
Ah. I was hoping that `setScale()` could be avoided but looking at statement that is not possible.
But as per SO:

https://stackoverflow.com/questions/1078953/check-if-bigdecimal-is-integer-value

what can be done is something like this:

```java
private boolean isIntegerValue(BigDecimal bd) {
  return bd.signum() == 0 || bd.scale() <= 0 || bd.stripTrailingZeros().scale() <= 0;
}
```

which seems quite optimal. And in particular avoids `remainder()` which, being division, is probably quite expensive operation.

Reply:
That's a great solution and more optimal! I will change the code to this one.